### PR TITLE
[shopsys] all constants now have set their visibility

### PR DESCRIPTION
**Pull request by TomasLudvik**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Visibility has been in most cases set by @access annotation that has not been ideal.
|New feature| No 
|[BC breaks](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md)| Yes 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: packages/framework/src/Form/Admin/Advert/AdvertFormType.php [hunk 1]
```diff
@@ -24,10 +24,8 @@
 
 class AdvertFormType extends AbstractType
 {
-    /** @access protected */
-    const VALIDATION_GROUP_TYPE_IMAGE = 'typeImage';
-    /** @access protected */
-    const VALIDATION_GROUP_TYPE_CODE = 'typeCode';
+    public const VALIDATION_GROUP_TYPE_IMAGE = 'typeImage';
```
Comment (vitek-rostislav) [inline packages/framework/src/Form/Admin/Advert/AdvertFormType.php]: another inconsistency

File: packages/framework/src/Model/Administrator/Security/AdministratorFrontSecurityFacade.php [hunk 2]
```diff
@@ -11,9 +11,7 @@
 
 class AdministratorFrontSecurityFacade
 {
-    // same as in security.yml
```
Comment (vitek-rostislav) [inline packages/framework/src/Model/Administrator/Security/AdministratorFrontSecurityFacade.php]: I think this line should not be removed

File: packages/framework/src/Model/Feed/FeedExportFactory.php [hunk 3]
```diff
@@ -10,11 +10,6 @@
 
 class FeedExportFactory
 {
-    /** @access protected */
```
Comment (vitek-rostislav) [inline packages/framework/src/Model/Feed/FeedExportFactory.php]: ok, makes sense as they are not used anywhere. However, I would expect this in another commit

File: packages/product-feed-google/src/Model/FeedItem/GoogleFeedItem.php [hunk 4]
```diff
@@ -9,7 +9,7 @@
 class GoogleFeedItem implements FeedItemInterface
 {
     protected const IDENTIFIER_TYPE_EAN = 'gtin';
-    protected const IDENTIFIER_TYPE_PARTNO = 'mpn';
+    public const IDENTIFIER_TYPE_PARTNO = 'mpn';
```
Comment (vitek-rostislav) [inline packages/product-feed-google/src/Model/FeedItem/GoogleFeedItem.php]: seems like unwanted change

File: docs/upgrade/UPGRADE-v8.0.0.md [hunk 5]
```diff
@@ -91,6 +91,8 @@ There you can find links to upgrade notes for other versions too.
     - the namespace of `FriendlyUrlDataProviderInterface` and `FriendlyUrlDataProviderRegistry` has changed from `Shopsys\FrameworkBundle\Component\Router\FriendlyUrl\CompilerPass` to `Shopsys\FrameworkBundle\Component\Router\FriendlyUrl` so change all your usages accordingly
     - you no longer need to tag your `FriendlyUrlDataProviders` with `shopsys.friendly_url_provider` because it is now done automatically
     - remove the usages of `RegisterFriendlyUrlDataProviderCompilerPass` class and `FriendlyUrlDataProviderRegistry::registerFriendlyUrlDataProvider` method because they have been removed
+- run `php phing phpstan` in order to check, that you are not using any private or protected constant ([#1181](https://github.com/shopsys/shopsys/pull/1181))
```
Comment (vitek-rostislav) [inline docs/upgrade/UPGRADE-v8.0.0.md]:
```suggestion
- run `php phing phpstan` in order to check, that you are not using any private or protected constant from outside of the class where the constant is declared ([#1181](https://github.com/shopsys/shopsys/pull/1181))
```

File: docs/upgrade/UPGRADE-v8.0.0.md [hunk 6]
```diff
@@ -91,6 +91,8 @@ There you can find links to upgrade notes for other versions too.
...
+    - if you have been using `NoUselessAccessFixer` or `ConstantVisibilityRequiredSniff` beware that these coding standards and has been replaced by `VisibilityRequiredFixer`
```
Comment (vitek-rostislav) [inline docs/upgrade/UPGRADE-v8.0.0.md]:
```suggestion
    - `Shopsys\CodingStandards\CsFixer\Phpdoc\NoUselessAccessFixer` has been removed, if you have been using it, you can use `PhpCsFixer\Fixer\Phpdoc\PhpdocNoAccessFixer` instead
    - `Shopsys\CodingStandards\Sniffs\ConstantVisibilityRequiredSniff` has been removed, if you have been using it, use `VisibilityRequiredFixer` with the following setting instead:
    ```
	PhpCsFixer\Fixer\ClassNotation\VisibilityRequiredFixer:
        elements:
            - 'property'
            - 'method'
            - 'const'
	```
```

File: packages/framework/src/Component/Router/Security/RouteCsrfProtector.php [hunk 7]
```diff
@@ -14,8 +14,7 @@
 class RouteCsrfProtector implements EventSubscriberInterface
 {
     public const CSRF_TOKEN_REQUEST_PARAMETER = 'routeCsrfToken';
-    /** @access protected */
-    const CSRF_TOKEN_ID_PREFIX = 'route_';
+    public const CSRF_TOKEN_ID_PREFIX = 'route_';
```
[the remark this answers is not in the document]
Comment (TomasLudvik) [reply packages/framework/src/Component/Router/Security/RouteCsrfProtector.php]: I have added info into commit message

File: packages/framework/src/Form/Admin/Advert/AdvertFormType.php (hunk 1 above)
Comment (TomasLudvik) [reply packages/framework/src/Form/Admin/Advert/AdvertFormType.php]: I have added info into commit message

File: packages/framework/src/Model/Administrator/Security/AdministratorFrontSecurityFacade.php (hunk 2 above)
Comment (TomasLudvik) [reply packages/framework/src/Model/Administrator/Security/AdministratorFrontSecurityFacade.php]: Reverted

File: packages/framework/src/Model/Feed/FeedExportFactory.php (hunk 3 above)
Comment (TomasLudvik) [reply packages/framework/src/Model/Feed/FeedExportFactory.php]: I have added info into commit message

File: packages/product-feed-google/src/Model/FeedItem/GoogleFeedItem.php (hunk 4 above)
Comment (TomasLudvik) [reply packages/product-feed-google/src/Model/FeedItem/GoogleFeedItem.php]: Reverted

File: docs/upgrade/UPGRADE-v8.0.0.md (hunk 5 above)
Comment (TomasLudvik) [reply docs/upgrade/UPGRADE-v8.0.0.md]: Done

File: docs/upgrade/UPGRADE-v8.0.0.md (hunk 6 above)
Comment (TomasLudvik) [reply docs/upgrade/UPGRADE-v8.0.0.md]: Done

**Pull request merged**